django/db/models/base.py

```diff
@@ -669,7 +669,7 @@ def get_deferred_fields(self):
         return {
             f.attname
             for f in self._meta.concrete_fields
-            if f.attname not in self.__dict__
+            if f.attname not in self.__dict__ and not f.generated
```

Comment:
I think this fails to account for cases where a generated field is explicit deferred.

For example,

```python
square = Square.objects.defer("area").get(pk=square_pk)
assert "area" in square.get_deferred_fields()
```

Reply:
Concretely it could break things like

```python
square = Square.objects.defer("area").get(pk=square_pk)
square.refresh_from_db()
```

As now `area` would be refreshed from db while deferred fields normally aren't.

Reply:
It looks like the right injection point but I think we need a specialized version that exclude generated fields for the `Model.save` purpose.